interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/bind-to-data/data-binding-modes.md

```diff
@@ -1 +1,118 @@
-[comment]: <> (This topic is intended for the Conversion Dialog. Explain when a user should select to convert its report from data bindings to expressions)
\ No newline at end of file
+---
+title: Data Binding Modes
+author: Anna Gubareva
+---
+# Data Binding Modes
+
+## Conversion Dialog
+
+The following dialog can appear when **expression bindings** are enabled in the Report Designer, and you [open an existing report](../open-reports.md) that uses standard **data bindings**:
+
+![](../../../../images/eurd-win-bindings-to-expressions-conversion-dialog.png)
+
+This dialog prompts you to convert your report to use expressions (the new binding mechanism). Click **Yes** to run the report conversion, click **No** to open the report without changes.
+
+If this dialog does not appear, check which binding mode the Report Designer uses.
+
+## Data Binding Modes
+
+* **Expression Bindings**
+
+    Enable you to use complex [expressions](../use-expressions.md) that include two or more fields and various functions. Expressions also allow you to calculate complex summaries without scripts and conditionally shape your data without formatting rules.
+
+    This mode is enabled in the Report Designer if the [Property Grid](../report-designer-tools/ui-panels/property-grid.md) provides the **Expressions** ![](../../../../images/eurd-win-property-grid-expressions-icon.png) tab.
+
+    ![](../../../../images/eurd-win-property-grid-expression-bindings-mode.png)
+
+* **Data Bindings**
+
+    Enable you to assign a single data field to a report control or use [report scripts](../use-report-scripts.md) in complex scenarios.
```

Comment:
complex scenarios?

Reply:
Let's write in the following way?
... or use report scripts to provide any custom logic to data bindings.

Reply:
use report scripts to provide custom logic to data bindings.